File: rolemadelen/programmers/120890_가까운_수.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(vector<int> array, int n) {
    int answer = 999;
    int min = 999;
    
    for(int i=0; i<array.size(); ++i) {
        int temp = abs(array[i] - n);
        if(temp == min) {
            answer = answer <= array[i] ? answer : array[i];
        } else if(temp < min) {
            min = temp;
            answer = array[i];
        }
    }
    
    return answer;
}

File: simsu/2주차/120890_가까운_수.js
function solution(array, n) {
    let compare = 101;
    let answer = 0;
    array.sort((a,b) => a-b);
    array.forEach(x => {
        if(compare > Math.abs(x-n)) {
            compare = Math.abs(x-n);
            answer = x;
        }
    })
    return answer;
}